### PR TITLE
Fix base image format generation, enable it for new image system, simplify

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | Read below
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | 
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/31488
| Related PRs       | 
| Sponsor company   |

### Description
- Multiple issues arose when implementing the image generation. This should fix everything.
- The new image system broke image generation with the old one. Regardless on what was set in `PS_IMAGE_QUALITY` it was always generating a JPG file and completely ignoring what was inside the file.
- It didn't work at all on legacy product page for example.
- Also, this PR enables to choose the base format for files with JPG extension when using the new generation system.
- I fixed the multiple images generation everywhere where it was previously implemented. If it was not implemented, I didn't implement it, because it would be generated in FO automatically.

### Settings with new image system
When enabling new image system, the settings are changed a bit. You can now configure the format for base PNG images and the new checkboxes.
newimagesettings

### Summary
- Old system - we generate:
  - JPG file with JPG/PNG/WEBP image inside
- New system - we generate:
  - JPG file with JPG/PNG image inside
  - optionally WEBP file with WEBP inside
  - optionally AVIF file with AVIF inside

### How to test - GENERAL
- To easily test it, download https://www.xnview.com/en/xnconvert/ and just drop whole folders into there and you will immediately see the file formats inside.
- Test generation when uploading on old product page.
- Test generation when uploading on new product page.
- Test manual regeneration in backoffice.
- Test automatic generation in front office. Just delete thumbnails and visit the product, they should be regenerated automatically.
- Product images should automatically generate on upload, but others may not. 
- The main thing to check that it generates nothing **WRONG** in backoffice. If it generates only jpg **PROPERLY** (with correct format inside) and the rest on the fly in front office, it's OK.

### How to test - OLD IMAGE SYSTEM
- Disable new multiple image generation feature flag.
- Set image settings to `Use PNG only if the base image is in PNG format`.
- Make a product, upload one JPG and one PNG image.
- Go to FO to this product to generate the thumbnails.
- Go to img/p folder and find the folder with images. Drop them all into XNCONVERT or simmilar software, or anything that will let you see what is inside. See that the thumbnail contain proper files.
- Then you can try again with "Use PNG" or "Use JPG" options. Just delete all the thumbnails and refresh the product in FO, so you don't have to create it again.

### How to test - NEW IMAGE SYSTEM
- Enable new multiple image generation feature flag.
- Set image settings to `Use PNG only if the base image is in PNG format` and enable webp AND/OR AVIF.
- Make a product, upload one JPG and one PNG image.
- Go to FO to this product to generate the thumbnails.
- Go to img/p folder and find the folder with images. Drop them all into XNCONVERT or simmilar software, or anything that will let you see what is inside. See that the thumbnail contain proper files. It will look something like this:
filetypes

### Known issues
- If you are on old image system and set image format to "WEBP" or "webp when original image is webp", then go enable new image system, when you go to image settings, you may not see the selected value. Everything will work, but you will get two WEBP images, one with jpg extension, one with webp extension. This can get fixed in another PR, I didn't want to mess with the experimental feature enabling.

### CODE REVIEW

File: src/Adapter/Image/ImageRetriever.php [hunk 1]
```diff
@@ -206,42 +212,23 @@ public function getImage($object, $id_image)
...
-                    }
-
-                    if ($generateHighDpiImages) {
-                        $this->checkOrGenerateImageType($originalImagePath, $imageFolderPath, $id_image, $image_type, $imageFormat, true);
-                    }
+                    $sources[$imageFormat] = $this->link->$getImageURL($rewrite, $id_image, $image_type['name'], $imageFormat);
                 }
             }
-
+
```
Comment (ghost) [inline src/Adapter/Image/ImageRetriever.php]:
```suggestion

```

**Comment by Hlavtox** [comment]
Ping @MatShir to confirm UI and wording. Old image system settings are unchanged.

**Review by zuk3975 — DISMISSED** [review]

**Comment by MatShir** [comment]
UI ✅ 
Wording need to validated with @l-delin but it is not blocking the PR 
Great work !

**Review by zuk3975 — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]
❤️

**Comment by l-delin** [comment]
Hello, everything's perfect - thanks!

**Comment by AureRita** [comment]
Hello @Hlavtox 

I tested your PR with all these scénarios : 


**With old image system :** 
- use png when : 
   - Test generation when uploading on old product page :heavy_check_mark: 
   - Test generation when uploading on new product page :heavy_check_mark: 
   - Test regenerate thumbnail :heavy_check_mark: 
   - test auto generation in front office :heavy_check_mark: 
   - check with category images  :heavy_check_mark: 

- use webp :
    - Test generation when uploading on old product page :heavy_check_mark: 
    - Test generation when uploading on new product page :heavy_check_mark: 
    - Test regenerate thumbnail :heavy_check_mark: 
    - test auto generation in front office :heavy_check_mark: 
   - check with category images  :heavy_check_mark: 
  
**With New image system** 
- with base/png only :
   - Test generation when uploading on old product page :heavy_check_mark: 
   - Test generation when uploading on new product page :heavy_check_mark: 
   - Test regenerate thumbnail :heavy_check_mark: 
   - test auto generation in front office :heavy_check_mark: 
   - check with category images  :heavy_check_mark: 

- with webp activated : 
   - Test generation when uploading on old product page :heavy_check_mark: 
   - Test generation when uploading on new product page :heavy_check_mark: 
   - Test regenerate thumbnail :heavy_check_mark: 
   - test auto generation in front office :heavy_check_mark: 
   - check with category images  :heavy_check_mark: 


So it can be Validated by QA

@MatShir  : I have a little problem about the UI, we already seen this type of UI with @nicosomb and we finally create this specs : https://github.com/PrestaShop/PrestaShop/issues/30944

This spec was create juste before the feature freeze in a meeting. 

If this PR is really validated, then we should change all the tests that have already been validated by the devs and yourself. 

Do we really change the UI ?

**Comment by Hlavtox** [comment]
@AureRita @MatShir The "design" in the issue did not take JPG/PNG option into consideration, so now it's named the "base image".

The settings page will be migrated to symfony for v9 I hope, so we can improve the wording. :-)

**Comment by MatShir** [comment]
Let's change the UI! Thanks @AureRita

**Comment by Hlavtox** [comment]
Aaaay, thanks for cooperation everybody!

**Comment by Robin-Fischer-PS** [comment]
@MatShir Why did you approve the PR ? It changes the UI, it's apparently not discussed (the new UI I mean), and QA will need to update once again the test...

**Comment by Hlavtox** [comment]
@Robin-Fischer-PS It changes the UI only after enabling the experimental flag.

**Comment by MatShir** [comment]
> So it can be Validated by QA

Tests were validated 
and also
I prefer taking a half day updating the specs and the tests than deprive the user a feature like this one behind a experimental feature flag

**Comment by Robin-Fischer-PS** [comment]
@MatShir The PR works, yes, but since it adds a load over QA, it would have been better to check with QA first before adding workload and putting a QA :heavy_check_mark: label.

**Comment by Hlavtox** [comment]
@Robin-Fischer-PS What load? :D @AureRita did a normal QA and just asked if the UI change is legitimate no?

**Comment by Hlavtox** [comment]
The feature sucked and needed to be changed. What does QA team has to say if an UI should be changed or not? If we decide to change it, it's your job to test it.

**Pull request merged**